main/app.js

```diff
@@ -1,6 +1,69 @@
-String.prototype.hasVowels = function() {};
-String.prototype.toUpper = function() {};
-String.prototype.toLower = function() {};
-String.prototype.ucFirst = function() {};
+String.prototype.hasVowels = function() {
+    return /[aeiou]/.test(this);
+};
+String.prototype.toUpper = function() {
+    var converted = '';
+    for (var index = 0; index < this.length; index++) {
+        converted += String.fromCharCode(this.charCodeAt(index) & 223);
+    }
+    return converted;
+};
+String.prototype.toLower = function() {
+    var converted = '';
+    for (var index = 0; index < this.length; index++) {
+        converted += String.fromCharCode(this.charCodeAt(index) | 32);
```

Comment:
Please add a comment here to explain what this line of code is doing and maybe consider splitting the implementation over two lines to make it more readable